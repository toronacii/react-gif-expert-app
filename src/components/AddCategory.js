import React, { useState } from 'react'
import PropTypes from 'prop-types';

export const AddCategory = ({ setCategories }) => {
  const [ value, setValue ] = useState('');

  return (
    <form onSubmit={handleSubmit}>
      <input
      type="text"
      value={value}
      onChange={handleChange} />
    </form>
  );

  function handleChange({ target }) {
    setValue(target.value);
  }

  function handleSubmit(e) {
    e.preventDefault();
    const category = value.trim();
    if (!category) return;

    setCategories(categories => {
      if (!categories.includes(category)) {
        setValue('');
        return [category, ...categories];
      }
      return categories;
    });
  }
}

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired
}
