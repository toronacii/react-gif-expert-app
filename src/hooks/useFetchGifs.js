import { useState, useEffect } from 'react';
import { getGifts } from '../helpers/getGifs';

export const useFetchGifs = (category) => {
  const [state, setState] = useState({
    gifs: [],
    loading: true
  });

  useEffect(() => {
    getGifts(category).then(gifs => setState({
      gifs,
      loading: false
    }));
  }, [category]);

  return state;
  
}