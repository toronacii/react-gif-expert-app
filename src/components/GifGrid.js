import PropTypes from 'prop-types';
import React from 'react';
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({ category }) => {

  const { gifs, loading } = useFetchGifs(category);

  return <>
    <h3 className="animate__animated animate__flash">{category}</h3>
    { loading && <div className="animate__animated animate__flash">Cargando...</div> }
    <div className="card-grid">
      {gifs.map(({ id, ...props }) => (
        <GifGridItem
          key={id}
          {...props} />)
      )}
    </div>
  </>
}

GifGrid.propTypes = {
  category: PropTypes.string.isRequired
}
