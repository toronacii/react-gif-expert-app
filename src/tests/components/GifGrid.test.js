import { shallow } from 'enzyme';
import React from 'react';
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';

jest.mock('../../hooks/useFetchGifs');

describe('Tests on `GifGrid` component', () => {
  const category = 'Dragon Ball';
  let wrapper = _createWrapper();

  beforeEach(() => {
    wrapper = _createWrapper();
  });

  test('GifGrid should be created', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should shows images when useFetchGifs is called', () => {
    const gifs = [{ id: 'abcdef', url: 'http://test', title: 'title' }];
    wrapper = _createWrapper(gifs, false);

    expect(wrapper.find('p').exists()).toBe(false);
    expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
    
  });

  function _createWrapper(gifs = [], loading = true) {
    useFetchGifs.mockReturnValue({ gifs, loading });
    return shallow(<GifGrid category={category} />);
  }
});