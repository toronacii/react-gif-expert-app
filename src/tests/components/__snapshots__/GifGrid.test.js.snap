// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on \`GifGrid\` component GifGrid should be created 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__flash"
  >
    Dragon Ball
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Cargando...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
